src/vm_pages.rs

```diff
@@ -616,7 +615,7 @@ impl<'a, T: GuestStagePagingMode, M> VmPagesMapper<'a, T, M> {
             .checked_add_pages(num_pages)
             .ok_or(Error::AddressOverflow)?;
         let regions = vm_pages.regions.read();
-        if !regions.contains(page_addr, end, region_type) {
+        if !regions.contains(page_addr, end, |r| r == region_type) {
             return Err(Error::InvalidMapRegion);
         }
         let mapper = vm_pages
```

Comment:
per our offline conversation, i think we should allow page faults in `*Removable` regions to continue to be serviceable as if they were the original region type. in other words:
- page faults taken in `*Removable` regions should not be fatal
- host should be allowed to add shared pages to `SharedRemovable` regions, and confidential zero pages to `ConfidentialRemovable`.

Reply:
Sounds good! What about measured pages?

Reply:
Please see the new commit handling this: https://github.com/rivosinc/salus/pull/261/commits/7c997b79e15b00b9a701a501286145f07b947054

Reply:
measured pages can only be added prior to tvm finalization, so it's impossible for those to be in a `*Removable` region anyway (which you've handled correctly.)